pkg/proc/interface.go

```diff
@@ -14,10 +10,21 @@ import (
 type Process interface {
 	Info
 	ProcessManipulation
-	BreakpointManipulation
 	RecordingManipulation
 }
 
+// ProcessInternal holds a set of methods that are
+// not meant to be called by anyone except for an instance of
+// `proc.Target`. These methods are not safe to use by themselves
+// and should never be called directly outside of the `proc` package.
+// This is temporary and in support of an ongoing refactor.
+type ProcessInternal interface {
+	WriteBreakpointFn(addr uint64) (string, int, *Function, []byte, error)
+	ClearBreakpointFn(uint64, []byte) error
+	AdjustsPCAfterBreakpoint() bool
+	CurrentDirection() Direction
```

Comment:
Restart, SetSelectedGoroutine, ContinueOnce and Detach should also be moved here.